pkg/bigquery/insert.go

```diff
@@ -0,0 +1,57 @@
+package bigquery
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type DataSaver struct {
+	Schema  bigquery.Schema
+	DataMap map[string]bigquery.Value
+}
+
+func (v DataSaver) Save() (row map[string]bigquery.Value, insertID string, err error) {
+	return v.DataMap, bigquery.NoDedupeID, nil
+}
+
+func insertDataToBigQuery(projectID, datasetID, tableName string, schema bigquery.Schema, valueSaver DataSaver, client *bigquery.Client) error {
+	ctx := context.Background()
+	tableRef := client.Dataset(datasetID).Table(tableName)
+	// Create the table if it doesn't exist
+	_, err := tableRef.Metadata(ctx)
```

Comment:
I'm thinking we can have a new interface `Setup()` to do the table creation. The `Setup()` function will be triggered when the user `connect` the connector

Reply:
As decided, we have moved the schema to resource and removed the ability to create table.